File: src/lchika.js
"use strict";

const gpio = require("gpio");

let isRunning = 1;
let setId;

const gpio4 = gpio.export(4, {
  direction: "out",
  ready(){
    let value = 1;
    setId = setInterval(() => {
      this.set(value);
      value++;
      value %= 2;
    }, 50);
  }
});

process.on("SIGINT", () => {
  gpio4.set(0);
  clearInterval(setId);
  gpio4.unexport();
  console.log("\nBye!!");
});

File: src/lchika.pl
use strict;
use warnings;
use utf8;
use Carp;
use 5.020;
use RaspberryPi::GPIO;
use Time::HiRes;
use AnyEvent;

main();

sub main {
  my $cv = AE::cv;

  my $gpio = RaspberryPi::GPIO->new({
    channel => 17,
    direction => "out",
    verbose => 1
  });

  my $value = 0;
  my $w; $w = AE::timer 0, 0.5, sub {
    my $v = $value % 2;
    $gpio->set($v);
    $value = ++$v;
  };

  my $finalize = AE::signal "INT", sub {
    print "\n";
    $gpio->unexport();
    $cv->send;
    undef $w;
  };

  $cv->recv;
  undef $finalize;
  say "Bye!!";
}

